daemon/routes/iprule/iprule.go

```diff
@@ -80,16 +81,72 @@ func (r *Router) SetupRoutingRules(
 			return err
 		}
 
-		otherRuleID, err := calculateRulePriority(ipv6)
-		if err != nil {
-			return err
+		// PeerA (LAN-a 192.168.1.x) connects to PeerB (LAN-b 192.168.1.x)
+		// if PeerB allows its LAN access when used as Exit node
+		// then PeerB LAN access is the priority over PeerA LAN
+		if !routeToRemoteLan {
```

Comment:
The code is duplicated with PreventLocalTraffic. I think it would be possible to refactor in a way that this would just call preventLocalTraffic.